RuntimeException: java.lang.NumberFormatException

## steps to reproduce
when configured to sync with planet Version 0.13.77, log in as a registered user and click on Library

```
 org.ole.planet.myplanet              E  FATAL EXCEPTION: main
  Process: org.ole.planet.myplanet, PID: 26843
  java.lang.RuntimeException: java.lang.NumberFormatException: For input string: ""
  at org.ole.planet.myplanet.ui.courses.AdapterCourses.onBindViewHolder(AdapterCourses.kt:141)
  at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7747)
  at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7847)
  at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6646)
  at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6917)
  at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6757)
  at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6753)
  at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
  at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
  at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
  at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
  at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4645)
  at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4348)
  at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4919)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
  at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
  at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:919)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
  at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
  at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
  at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:919)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1263)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
  at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
2024-05-02 10:20:45.796 26843-26843 AndroidRuntime          org.ole.planet.myplanet              E    at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
  at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
  at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
  at com.android.internal.policy.DecorView.onLayout(DecorView.java:789)
  at android.view.View.layout(View.java:24421)
  at android.view.ViewGroup.layout(ViewGroup.java:6440)
  at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4258)
  at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3583)
  at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2465)
  at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9305)
  at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1339)
  at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1348)
  at android.view.Choreographer.doCallbacks(Choreographer.java:952)
  at android.view.Choreographer.doFrame(Choreographer.java:882)
  at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1322)
  at android.os.Handler.handleCallback(Handler.java:958)
  at android.os.Handler.dispatchMessage(Handler.java:99)
  at android.os.Looper.loopOnce(Looper.java:205)
  at android.os.Looper.loop(Looper.java:294)
  at android.app.ActivityThread.main(ActivityThread.java:8177)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
                                                                                                    Caused by: java.lang.NumberFormatException: For input string: ""
  at java.lang.Long.parseLong(Long.java:761)
  at java.lang.Long.parseLong(Long.java:876)
  at org.ole.planet.myplanet.ui.courses.AdapterCourses.onBindViewHolder(AdapterCourses.kt:139)
  ... 86 more
---------------------------- PROCESS ENDED (26843) for package org.ole.planet.myplanet ----------------------------
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -9,8 +9,8 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 21
         targetSdkVersion 34
-        versionCode 1510
-        versionName "0.15.10"
+        versionCode 1511
+        versionName "0.15.11"
         ndkVersion '21.3.6528147'
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true

--- a/app/src/main/java/org/ole/planet/myplanet/model/RealmMyLibrary.kt
+++ b/app/src/main/java/org/ole/planet/myplanet/model/RealmMyLibrary.kt
@@ -50,7 +50,7 @@ open class RealmMyLibrary : RealmObject() {
     @JvmField
     var uploadDate: String? = null
     @JvmField
-    var createdDate: String? = null
+    var createdDate: Long = 0
     @JvmField
     var openWith: String? = null
     @JvmField
@@ -399,7 +399,7 @@ open class RealmMyLibrary : RealmObject() {
             resource?.openWith = JsonUtils.getString("openWith", doc)
             resource?.articleDate = JsonUtils.getString("articleDate", doc)
             resource?.kind = JsonUtils.getString("kind", doc)
-            resource?.createdDate = JsonUtils.getLongAsString("createdDate", doc)
+            resource?.createdDate = JsonUtils.getLong("createdDate", doc)
             resource?.language = JsonUtils.getString("language", doc)
             resource?.author = JsonUtils.getString("author", doc)
             resource?.mediaType = JsonUtils.getString("mediaType", doc)

--- a/app/src/main/java/org/ole/planet/myplanet/ui/resources/AdapterResource.kt
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/resources/AdapterResource.kt
@@ -83,7 +83,7 @@ class AdapterResource(private val context: Context, private var libraryList: Lis
                     String.format("%.1f", libraryList[position]?.averageRating!!.toDouble())
                 }
             Log.d("AdapterLibrary", "onBindViewHolder: ${libraryList[position]?.createdDate}")
-            holder.rowLibraryBinding.tvDate.text = libraryList[position]?.createdDate?.toLong()?.let { formatDate(it, "MMM dd, yyyy") }
+            holder.rowLibraryBinding.tvDate.text = libraryList[position]?.createdDate?.let { formatDate(it, "MMM dd, yyyy") }
             displayTagCloud(holder.rowLibraryBinding.flexboxDrawable, position)
             holder.itemView.setOnClickListener { openLibrary(libraryList[position]) }
             holder.rowLibraryBinding.ivDownloaded.setImageResource(

--- a/app/src/main/java/org/ole/planet/myplanet/ui/resources/AddResourceActivity.kt
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/resources/AddResourceActivity.kt
@@ -97,7 +97,7 @@ class AddResourceActivity : AppCompatActivity() {
         resource.subject = subjects
         resource.setUserId(RealmList())
         resource.level = levels
-        resource.createdDate = Calendar.getInstance().timeInMillis.toString()
+        resource.createdDate = Calendar.getInstance().timeInMillis
         resource.resourceFor = resourceFor
         resource.resourceLocalAddress = resourceUrl
         resource.resourceOffline = true

